apps/dav/lib/Connector/Sabre/File.php

```diff
@@ -225,7 +225,7 @@ public function put($data) {
 			if (isset($_SERVER['CONTENT_LENGTH']) && $_SERVER['REQUEST_METHOD'] === 'PUT') {
 				$expected = (int)$_SERVER['CONTENT_LENGTH'];
 				if ($count !== $expected) {
-					throw new BadRequest('expected filesize ' . $expected . ' got ' . $count);
+					throw new BadRequest('Expected filesize of ' . $expected . ' bytes but read (from Nextcloud client) and wrote (to Nextcloud storage) ' . $count . ' bytes. Could either be a network problem on the sending side or a problem writing to the storage on the server side.');
```

Comment:
```suggestion
					throw new BadRequest('Expected filesize of ' . $expected . ' bytes but read (from client) and wrote (to Nextcloud storage) ' . $count . ' bytes. Could either be a network problem on the sending side or a problem writing to the storage on the server side.');
```

because true for any other web dav client.

Reply:
This was more meant as „it’s a client to Nextcloud“ - Same for the server part, because the backend can involve multiple servers.